xbmc/Application.cpp

```diff
@@ -3489,8 +3496,20 @@ void CApplication::OnPlayerCloseFile(const CFileItem &file, const CBookmark &boo
   CFileItem fileItem(file);
   CBookmark resumeBookmark;
   bool playCountUpdate = false;
+  float percent = 0.0f;
 
-  float percent = bookmark.timeInSeconds / bookmark.totalTimeInSeconds * 100;
+  if (IsPlayingRegularStack())
+  {
+    // for stacks, we have to save the bookmark on the stack, held in m_itemCurrentFile
+    fileItem = *m_itemCurrentFile;
```

Comment:
this is not correct. stopping player is asynchronous and m_itemCurrentFile may have changed at the time this method is called. this is why file is passed as a paramter to this method.

Reply:
You're right, unless m_itemCurrentFile should really reflect the currently playing file for the entire duration of playback (from OnPlayBackStarted to OnPlayerCloseFile, see my other comment as well). I'll think about another way.